File: code_samples/lua/data_structures_linked_list.lua
-- Data Structures: Linked List
-- AI/ML Training Sample

LinkedList = {}
LinkedList.__ index = LinkedList

function LinkedList:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function LinkedList:process(input)
    self.data = input
end

function LinkedList:getData()
    return self.data
end

function LinkedList:validate()
    return #self.data > 0
end

-- Example usage
local instance = LinkedList:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/typescript/data_structures_linked_list.ts
/**
 * Data Structures: Linked List
 * AI/ML Training Sample
 */

interface ILinkedList {
    data: string;
    process(input: string): void;
    validate(): boolean;
}

class LinkedList implements ILinkedList {
    data: string;
    
    constructor() {
        this.data = "";
    }
    
    process(input: string): void {
        this.data = input;
    }
    
    getData(): string {
        return this.data;
    }
    
    validate(): boolean {
        return this.data.length > 0;
    }
}

// Example usage
const instance = new LinkedList();
instance.process("example");
console.log(`Data: ${instance.getData()}`);
console.log(`Valid: ${instance.validate()}`);

export { LinkedList, ILinkedList };
